plugin/file/lookup.go

```diff
@@ -197,11 +197,22 @@ func (z *Zone) Lookup(ctx context.Context, state request.Request, qname string)
 
 	}
 
+	ce, foundCe = z.ClosestEncloser(qname)
+
 	// Haven't found the original name.
 
 	// Found wildcard.
 	if wildElem != nil {
 		auth := ap.ns(do)
+		if foundCe && dns.CountLabel(ce.Name())+1 > dns.CountLabel(wildElem.Name()) {
+			// wildcard denial
```

Comment:
this comment is a bit terse to explain the previous if statement, please expand a bit

Reply:
This is the case described in https://github.com/coredns/coredns/issues/4290, when the record "something.r.c.d.example.com" is matching "*.d.example.com" wildcard, "r.c.d.example.com", its closest encloser (CE) can be found, however, its CE is not directly under the wildcard (cntLabel(CE) + 1 > cntLabel(wildcard)), the record needs to be resolved as nxdomain.